e2e/nomostest/predicates.go

```diff
@@ -208,16 +209,16 @@ func HasCorrectResourceRequestsLimits(containerName string, cpuRequest, cpuLimit
 		}
 		for _, container := range dep.Spec.Template.Spec.Containers {
 			if containerName == container.Name {
-				if container.Resources.Requests[corev1.ResourceCPU] != cpuRequest {
+				if !equality.Semantic.DeepEqual(container.Resources.Requests[corev1.ResourceCPU], cpuRequest) {
```

Comment:
I'm a bit confused, is this a bug in our code or e2e tests? Also, this seems like something that should be unit-testable. Do we have coverage for this in our e2e tests?

Reply:
Just saw the description so I see this was fixed in another PR. I'm still curious about unit testing though

Reply:
> I'm still curious about unit testing though

We can add some unit test coverage into pkg/reconcilermanager/controller.
Are you recommending to avoid testing this in the e2e test completely?

Reply:
A validation such as this seems like a better fit for lightweight unit tests. It doesn't seem like we should need e2e tests for this IMO, but I haven't looked too closely at the related code. We can also get more thorough scenario coverage through unit tests

Reply:
This PR does not add new test cases. It just adds more validations into two existing e2e testcases.